libcudacxx/include/cuda/std/__type_traits/is_any_floating_point.h

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___TYPE_TRAITS_IS_ANY_ARITHMETIC_H
+#define _LIBCUDACXX___TYPE_TRAITS_IS_ANY_ARITHMETIC_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__type_traits/is_extended_floating_point.h>
+#include <cuda/std/__type_traits/is_floating_point.h>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+template <class _Tp>
+inline constexpr bool __is_any_floating_point_v =
+  _CCCL_TRAIT(is_floating_point, _Tp) || _CCCL_TRAIT(__is_extended_floating_point, _Tp);
```

Comment:
I am not really in favor of that. I believe it is important to be explicit what is supported by extended floating point types and represent that in code. Also if we decide to do that we should simplify this to:
```suggestion
inline constexpr bool __is_any_floating_point_v = is_floating_point_v<_Tp> || __is_extended_floating_point_v<_Tp>;
```

Reply:
I just need a simple type trait to identify any floating point, e.g.
```cpp
if constexpr (__is_any_floating_point_v<T>)
    return T{2.5f};
else
    return T{1};
```